In [1]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

In [2]:
df=pd.read_excel('D:\PGP IN DATA SCIENCE with Careerera\Data Sets\innomatics\scores_data.xlsx')
df.head()

batch  user_id             Score     score
0  AI_ELITE_7  uid_149  2023-06-07 00:00:00      6
1  AI_ELITE_7  uid_148  2023-06-07 00:00:00      6
2  AI_ELITE_7  uid_147  2023-07-07 00:00:00      7
3  AI_ELITE_7  uid_146  2023-07-07 00:00:00      7
4  AI_ELITE_7  uid_145  2023-04-07 00:00:00      4

In [3]:
df.columns

Index(['batch', 'user_id', '   Score   ', 'score'], dtype='object')

In [4]:
df.drop('   Score   ', axis=1, inplace=True)
df.head()

batch  user_id  score
0  AI_ELITE_7  uid_149      6
1  AI_ELITE_7  uid_148      6
2  AI_ELITE_7  uid_147      7
3  AI_ELITE_7  uid_146      7
4  AI_ELITE_7  uid_145      4

In [7]:
df.dtypes

batch      object
user_id    object
score       int64
dtype: object

### Total Batches?

In [5]:
df.batch.value_counts()

AI_ELITE_7    53
AI_ELITE_6    48
AI_ELITE_4    48
Name: batch, dtype: int64

**There are 3 batches. batch  AI_ELITE_4, AI_ELITE_6 has 48 students and AI_ELITE_7 has 53 students.**

### Average score by each batch

In [6]:
df.groupby('batch').score.mean()

batch
AI_ELITE_4    3.791667
AI_ELITE_6    4.229167
AI_ELITE_7    5.056604
Name: score, dtype: float64

**Average score by AI_ELITE_4 batch is low ie just 3.8. and the average score by batch AI_ELITE_7 is high ie about 5.05.**

### Distribution of scores

In [7]:
batch_4=df[df['batch']=='AI_ELITE_4']
batch_6=df[df['batch']=='AI_ELITE_6']
batch_7=df[df['batch']=='AI_ELITE_7']

### Batch 4

In [36]:
print(batch_4.batch.count())
print(batch_4['score'].mean())
print(batch_4['score'].std())

48
3.7916666666666665
1.4433756729740639


In [10]:
batch_4['score'].value_counts()

4    19
5     8
3     8
2     7
7     3
6     1
1     1
0     1
Name: score, dtype: int64

In [23]:
19/48*100

39.58333333333333

In [24]:
42/48*100

87.5

**19 out of 48 students, scored 4 marks out of 7 marks.**

### Batch 6

In [37]:
print(batch_6.batch.count())
print(batch_6['score'].mean())
print(batch_6['score'].std())

48
4.229166666666667
1.6404408246842614


In [11]:
batch_6['score'].value_counts()

5    13
4    11
3     9
7     5
6     4
2     3
1     2
0     1
Name: score, dtype: int64

In [27]:
23/48*100

47.91666666666667

**Most of the Students obtained more than 2 marks.**

### Batch 7

In [38]:
print(batch_7.batch.count())
print(batch_7['score'].mean())
print(batch_7['score'].std())

53
5.056603773584905
1.4466818046245138


In [12]:
batch_7['score'].value_counts()

6    13
5    11
7    10
4    10
3     7
2     2
Name: score, dtype: int64

In [28]:
51/53*100

96.22641509433963

**There are no students who obtained less than 2 marks. Most of the students marks are in the range of 3 to 7.**

## Inference:
### `1. There are 3 batches, AI_ELITE_4, AI_ELITE_6 and AI_ELITE_7.`

### `2. Batch AI_ELITE_4 and AI_ELITE_6 has 48 students each and AI_ELITE_7 has 53 students.`

### `3. Average score of AI_ELITE_4 batch is low ie just 3.8 marks, average score of AI_ELITE_6 is 4.22 marks and the average score of batch AI_ELITE_7 is high ie about 5.05 marks out of 7 marks.`

### `4. 19 out of 48 students of batch 4 ie 39.6% students scored 4 marks out of 7 marks. Scores of  88% students in batch 4 is in the range from 2 to 5 marks.`

### `5. 48% Students in batch 6 obtained marks between 3 and 5.`

### `6. In the batch 7 None of the student got Zero and One marks, range of marks obtained by most of the students(96%) from 3 to 7, hence the average is high for this batch.`

### `7. Students of Batch 7 are performing better than the rest two batches.`